### ISSUE TITLE
AnimationNodeAnimation in AnimationTree should have configurable start and end times

### ISSUE THREAD

**Issue by Cwiiis** [godotengine/godot-proposals#1121]
**Describe the project you are working on:**
I'm working on a 2.5d scrolling arcade shooter/platformer (2d gameplay, 3d environment and physics).

**Describe the problem or limitation you are having in your project:**
Many of the resources I receive have one single animation, with frame-ranges specified for particular sub-animations. This is true for many of the resources available on both SketchFab and TurboSquid.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
An enhancement would be for the AnimationNodeAnimation in AnimationTree to let me specify start/end times instead of always playing the whole animation. To overcome the problem currently, I need to edit the model in Blender, which is laborious, non-trivial and error-prone. This feature would make this entirely unnecessary and would allow direct use of many resources from SketchFab and TurboSquid (and presumably other 3d model resource sites).

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Two float properties would be available on AnimationNodeAnimation, start and end. Values below zero for end would signify no end (so play to end of animation). Values would otherwise be interpreted as times, in the same way the input value for seeking is treated. These properties would be modifiable from code and modifiable in the editor via the inspector.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I've found no way to work around this limitation beyond manually editing animations. I believe this enhancement would be used often.

**Is there a reason why this should be core and not an add-on in the asset library?:**
This is a common property of many 3d models with animation and any 3d game involving animated models may find this feature useful. Its behaviour and use is obvious enough not to introduce friction in the case where you don't have use for this feature. This feature would also be useful if you need to make small tweaks to the start/end of animations without having to edit files or go back to an artist to edit files for you.

**Comment by iapps207** [comment]
I totally see sense in this as this is one of the problems am going through in issue #1105
So I think this actually gives more power to the animation player node

**Comment by wijat** [comment]
Yes, this is the big bug I meet, I can not control animation because of this issue!
AnimationNodeAnimation can not find AnimationNode method!
![图片](https://user-images.githubusercontent.com/8220843/93739579-38791f80-fc1b-11ea-97aa-acfb4d49d481.png)
